docs/design/satellite-svc-separation.md

```diff
@@ -0,0 +1,93 @@
+# Satellite Service Separation
+
+## Abstract
+
+The goal of the design doc is to describe the work necessary to make the Satellite horizontally scalable.
+
+## Background
+
+Currently all Satellite services run in a single binary in a single container. While this is great for development, this is bad for scaling.
+
+Currently the Satellite is a single binary made of up the following services:
+- version
+- overlay
+- metainfo
+- inspector
+- orders
+- repair (includes checker and repairer)
+- audit
+- garbage collection (GC)
+- accounting (includes tally and rollup service and the live accounting cache)
+- mail
+- console
+- nodestats
+- note: marketing service is currently on pause and is not included in this doc
+- note: kademlia, discovery, and vouchers are being removed and not included in this doc
+
+The Satellite has the following public GRPC endpoints:
+- metainfo
+- nodestats
+- orders
+
+The Satellite has the following privare GRPC endpoints:
+- inspectors
+
+All services (except version) make connections to the masterDB. Five services rely on metainfo service to access the metainfoDB, this includes inspectors, accounting, audit, repair, and garbage collection.
```

Comment:
is `masterDB` the name of one of our databases? (It doesn't appear to show up in a search in our code.) If not, which DB is meant here? The overlay db?

Reply:
i guess the docs call masterDB the Satellite.DB so I should change it here to be the dame. Though I'm confused about pointerDB versus metainfoDB.  I think we are moving toward metainfoDB so maybe I should update those docs too.  https://github.com/storj/docs/blob/master/code/persistentstorage/Databases.md#satellites